### PR TITLE
fix multiple abstracts ubtue_brill live https://github.com/ubtue/DatenProbleme/issues/1637

### PR DESCRIPTION
**Pull request by socheres**

### CODE REVIEW

File: ubtue_Brill.js [hunk 1]
```diff
@@ -70,13 +70,21 @@ function getSearchResults(doc) {
...
-	  item.abstractNote = ZU.xpath(doc, '//section[@class="abstract"]//p');
-	  if (item.abstractNote && item.abstractNote.length > 0)
-		 item.abstractNote = item.abstractNote[0].textContent.trim();
-	  else
-		 item.abstractNote = '';
+	let abstracts = ZU.xpath(doc, '//section[@class="abstract"]//p');
+	//multiple abstracts
+	if (abstracts && abstracts.length > 0) {
+		item.abstractNote = abstracts[0].textContent.trim().replace(/^(Résumé\s?)/i, "");
+		if (abstracts.length ) {
```
Comment (jriedl) [inline ubtue_Brill.js:77]: This is always true because of the condition in line 75.

File: ubtue_Brill.js [hunk 2]
```diff
@@ -70,13 +70,21 @@ function getSearchResults(doc) {
...
+			for (let abs of abstracts) {
```
Comment (jriedl) [inline ubtue_Brill.js:78]: The first abstract is added and afterwards compared again as to whether or not to add it again. It is probably better to slice the original `abstracts` structure after the first assignment (c.f. e.g. `https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/slice`) and then add the rest unconditionally.

File: ubtue_Brill.js (hunk 1 above)
Comment (socheres) [reply ubtue_Brill.js]: ok

File: ubtue_Brill.js (hunk 2 above)
Comment (socheres) [reply ubtue_Brill.js]: the abstracts.length is "4" with twice the same objects value. `Z.debug(abstracts.length)` Therefore I have to do this unusual if-comparison for deduplication.

Comment (jriedl) [reply ubtue_Brill.js]: Think this deduplication should be made explicit then:
````js
    let abstracts = ZU.xpath(doc, '//section[@class="abstract"]//p');
    //multiple abstracts
    if (abstracts && abstracts.length > 0) {
	// Deduplicate original results
	abstracts = [...new Set(abstracts.map(abstract => abstract.textContent))];
	item.abstractNote = abstracts[0].trim().replace(/^(Résumé\s?)/i, "");
	if (abstracts.length > 1) {
	    for (let abs of abstracts.slice(1)) {
                  item.notes.push({
	 	         note: "abs:" + ZU.trimInternal(abs),
	 	  });
            }
	}
    }

Comment (socheres) [reply ubtue_Brill.js]: ok. I didn't come up with that idea, but the one line code with arrow function looks good :+1: For my understanding
```
//arrow function
abstracts = [...new Set(abstracts.map(abstract => abstract.textContent))];

//regular function
abstracts = Array.from(new Set(abstracts.map(function(abstract) {
	return abstract.textContent
})));
```

Comment (jriedl) [reply ubtue_Brill.js]: These two function should yield identical results. The first one uses the spread operator `...` and an arrow notation as argument to map, so the first one is a "spread and arrow function" ;-).

Comment (jriedl) [reply ubtue_Brill.js]: Unfortunately, I probably cannot commit to your repository, so I cannot fix the formatting problems myself. I will merge it now but time permitting please fix them.

**Comment by jriedl** [comment]
In Kürze auf ptah.

**Comment by jriedl** [comment]
nu natürlich.

**Pull request merged**